apps/web/src/app/karriere/[id]/page.tsx

```diff
@@ -9,12 +13,153 @@ interface JobListingProps {
 }
 
 const JobListingPage = async ({ params }: JobListingProps) => {
-  const { id } = await params
-  const data = await server.jobListing.get.query(id)
-  if (!data) {
+  const { id: rawParamId } = await params
+  const paramId = decodeURIComponent(rawParamId)
+
+  const jobListing = await server.jobListing.get.query(paramId)
+
+  if (!jobListing) {
     return notFound()
   }
-  return <JobListingView jobListing={data} />
+
+  const applyButton = getApplyButton(jobListing)
+
+  return (
+    <div className="flex flex-col gap-4">
+      <Button
+        variant="text"
+        element={Link}
+        href="/karriere"
+        icon={<Icon icon="tabler:arrow-left" className="text-lg" />}
+        className="w-fit"
+      >
+        Andre muligheter
+      </Button>
+
+      <Title element="h1" size="xl" className="text-4xl">
+        {jobListing.title}
+      </Title>
+
+      <div className="flex flex-row gap-8 justify-between">
+        <div className="w-2/3">
+          <RichText content={jobListing.description} />
+        </div>
+
+        <div className="flex flex-col gap-4 w-1/3">
+          <CompanyBox company={jobListing.company} />
+
+          <ApplicationInfoBox jobListing={jobListing} />
+
+          {applyButton}
+        </div>
+      </div>
+    </div>
+  )
+}
+
+const ApplicationInfoBox = ({ jobListing }: { jobListing: JobListing }) => {
+  const employmentType = getEmploymentType(jobListing.employment)
+  const locationNames = jobListing.locations.map((location) => location.name).join(", ") || "Ingen lokasjon"
+  const deadlineElement = getDeadline(jobListing.deadline, jobListing.deadlineAsap)
+
+  return (
+    <div className="flex flex-col gap-4 p-6 bg-gray-100 rounded-2xl">
+      <Title size="md">Stillingsinformasjon</Title>
+
+      <div className="flex flex-row gap-2 items-center">
+        <Icon icon="tabler:world" width={20} height={20} />
+        <Text>{locationNames}</Text>
+      </div>
+
+      <div className="flex flex-row gap-2 items-center">
+        <Icon icon="tabler:clock" width={20} height={20} />
+        {deadlineElement}
+      </div>
+
+      <div className="flex flex-row gap-2 items-center">
+        <Icon icon="tabler:calendar-event" width={20} height={20} />
+        <div className="flex flex-col gap-0.5">
+          {jobListing.start && <Text>Starter {formatDate(jobListing.start, "dd. MMM yyyy")}</Text>}
+          {jobListing.end && <Text>Slutter {formatDate(jobListing.end, "dd. MMM yyyy")}</Text>}
+        </div>
+      </div>
+
+      <div className="flex flex-row gap-2 items-center">
+        <Icon icon="tabler:briefcase" width={20} height={20} />
+        <Text>{employmentType}</Text>
+      </div>
+    </div>
+  )
+}
+
+const CompanyBox = ({ company }: { company: JobListing["company"] }) => {
```

Comment:
Prefer not to inline object types, it hurts readability. Also feel free to just pass down the entire object.

```suggestion
interface CompanyBoxProps {
  listing: JobListing
}
const CompanyBox = ({ company }: CompanyBoxProps) => {
```